pkg/proxy/ipvs/proxier.go

```diff
@@ -1645,10 +1647,19 @@ func (proxier *Proxier) writeIptablesRules() {
 			}
 			args = append(args,
 				"-m", "comment", "--comment", proxier.ipsetList[set.name].getComment(),
+			)
+			if !set.jumpIfMatch {
+				args = append(args, "!")
+			}
```

Comment:
The "!" comes in the wrong place, it should be before "--match". Suggested;
```go
			if !set.jumpIfMatch {
				args = append(args,
					"-m", "set", "!", "--match-set", proxier.ipsetList[set.name].Name,
					set.matchType)
			} else {
				args = append(args,
					"-m", "set", "--match-set", proxier.ipsetList[set.name].Name,
					set.matchType)
			}
```

Reply:
In the same area; when a negated condition is used and empty set becomes _very_ significant. Extend the "ifEmpty" condition (or drop it entirely).